### ISSUE TITLE
External functions are not evaluated in order

### ISSUE THREAD

**Issue by ncthbrt** [inkle/ink#81]
The following code does not execute the way I would expect it to. Perhaps I am making a naive mistake, as I have just started using this framework this evening, but one would expect the code below to produce this output (please excuse the shitty writing, I wanted to simplify the example):

```text
   I didn't know who I was.
   "Who are you?" the blackness mockingly asked: Nick Cuthbert
   Nick Cuthbert was a good a name as any
```

Instead I am getting the  following:
```text
    "Who are you?" the blackness mockingly asked:  Nick Cuthbert
    I didn't know who I was

    Nick Cuthbert was a good name as any
```    

Code:

```ink
EXTERNAL prompt(promptString)
VAR playerName = "Earnest"

-> start

== start ==
I didn't know who I was
~ playerName = prompt ("\"Who are you?\" the blackness mockingly asked: ") 
{playerName} was a good name as any.

```

```C#
string story = File.ReadAllText("out.ink.json");

var inkStory = new Story(story);
inkStory.BindExternalFunction("prompt", (string text) =>
{
    Console.Write(text);
    return Console.ReadLine();
});
```

**Comment by joethephish** [comment]
I'm afraid this is working as intended. Ink builds up the final output gradually, and only once it's finished evaluating does the final text become available. It's not designed to be used in conjunction with `Console.Write()`. In addition, the external function support isn't designed for "interactive" functions either. They're designed for functions that return immediately.

So, you need to build this in such a way that you allow `Continue()` (or `ContinueMaximally()`) to complete before you allow player input. I'd suggest that you do something like create a dummy choice:

    VAR playerName = "Unnamed"

    I didn't know who I was.
    "Who are you? the blackness mockingly asked:
    * [PROMPT]
    - {playerName} was a good name as any.

Then in the game code, check for a choice with a particular title, and set the variable directly:

    story.variablesState["playerName"] = Console.ReadLine();

...and `Continue()`. (I haven't actually tried this, but it's the general approach I would take!)

**Comment by ncthbrt** [comment]
Thanks. I'll give that a try.
